File: examples/go/minimal-order-service/application/ports/inbound/place_order_port.go
package inbound

import "github.com/example/order-service/domain"

// PlaceOrderCommand represents the command to place an order.
type PlaceOrderCommand struct {
	Amount   int64
	Currency string
}

// PlaceOrderPort is the inbound port that defines how the application can be used.
type PlaceOrderPort interface {
	PlaceOrder(command PlaceOrderCommand) (*domain.Order, error)
}

File: examples/python/minimal-order-service/application/ports/inbound/place_order_port.py
"""
Inbound port: defines how the application can be used.
"""
from abc import ABC, abstractmethod
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from domain.order import Order


@dataclass
class PlaceOrderCommand:
    """Command DTO for placing an order."""
    amount: int  # in cents
    currency: str


class PlaceOrderPort(ABC):
    """
    Inbound port: defines how the application can be used.
    This is the interface that adapters (REST, CLI) will call.
    """
    
    @abstractmethod
    def place_order(self, command: PlaceOrderCommand) -> 'Order':
        """
        Place an order.
        
        Args:
            command: Place order command
        
        Returns:
            Created order
        """
        pass
